lib/tree.rb

```diff
@@ -17,41 +17,139 @@ def initialize
   end
 
   # Time Complexity: 
-  # Space Complexity: 
+  # Space Complexity: o(n)
   def add(key, value)
```

Comment:
The space/time complexity is O(log n) if the tree is balanced and O(n) if it's not.